File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Protocol/Parsers/DeletedResponseParser.cs
using VertexCache.Sdk.Results;

namespace VertexCache.Sdk.Protocol.Parsers
{
    public class DeletedResponseParser : IResponseParser
    {
        public bool CanParse(string response) => response == "+Deleted";

        public VCacheResult Parse(string response) => VCacheResult.SuccessWithRaw(response);
    }
}

File: client-sdk/go/sdk/protocol/parsers/deleted_response_parser.go
package parsers

import (
	"strings"

	"vertexcache/sdk/results"
)

type DeletedResponseParser struct{}

func (p *DeletedResponseParser) CanParse(line string) bool {
	return strings.HasPrefix(line, "+Deleted")
}

func (p *DeletedResponseParser) Parse(line string) *results.VCacheResult {
	return results.NewSuccessWithRaw(line)
}
